packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts

```diff
@@ -46,6 +63,74 @@ function validateKeyField(field: FieldDefinitionNode): void {
     throw new InvalidDirectiveError(`If you define a field and specify it as a 'keyField', it must be of type 'ID' or 'String'.`)
 }
 
+/**
+ * Ensure that the field passed in is compatible to be a key field
+ * (Not a list and of type ID or String)
+ * @param field: the field to be checked.
+ */
+function validateKeyFieldNewConnection(field: FieldDefinitionNode): void {
+    if (!field) {
+        return
+    }
+    const isNonNull = isNonNullType(field.type);
+    const isAList = isListType(field.type)
+
+    // The only valid key fields are single non-null fields.
+    if (!isAList && isNonNull) {
+        return;
+    }
+    throw new InvalidDirectiveError(`All fields used for a connection cannot be lists and must be of Non-Null types.`)
```

Comment:
Can we clean up this error msg to be more active. I.E. `All fields provided to a @connection must be non-null scalar fields.`